Source/WebCore/page/SecurityOriginData.h

```diff
@@ -34,11 +36,16 @@ class Frame;
 class SecurityOrigin;
 
 struct SecurityOriginData {
+    enum OpaqueOriginIdentifierType { };
+    using OpaqueOriginIdentifier = ProcessQualified<ObjectIdentifier<OpaqueOriginIdentifierType>>;
+    using MarkableOpaqueOriginIdentifier = Markable<OpaqueOriginIdentifier, OpaqueOriginIdentifier::MarkableTraits>;
+
     SecurityOriginData() = default;
-    SecurityOriginData(const String& protocol, const String& host, std::optional<uint16_t> port)
+    SecurityOriginData(const String& protocol, const String& host, std::optional<uint16_t> port, MarkableOpaqueOriginIdentifier opaqueOriginIdentifier = { })
```

Comment:
I think it would be less confusing to add a separate constructor that takes only an opaqueOriginIdentifier since the other parameters don't make sense when the origin is unique.

Reply:
Good point. We'll need something like this for the IPC decoder, but we can make this constructor private and leave the original constructor as public.